### ISSUE TITLE
UI: Add notices for published, updated, save failed

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#967]
Splitting this out from #707. The flow for publishing and saving is soon in a good place. 

We have, however, a need for a few edgecase notices:

- Published successfully
- Updated successfully
- Scheduled successfully

... as well as failure states:

- Publishing failed
- Updating failed
- Scheduling failed

... and also for draft saving and auto saving:

- Save failed

## Save Failed

Let's make this part of the "Save" area on the left:

![saving publishing](https://cloud.githubusercontent.com/assets/1204802/26670513/39fcff42-46b2-11e7-90e5-caa61aaf2e5d.png)

## Publish/Updating/Scheduling notices

We'll probably want notices after all (see also #486), because thankfully we were able to test for a while a design where the notice itself was integrated in the button. Not only did this feel ephemeral, but it made the button itself feel more complex. It also doesn't work that well if we ever get to exploring a dropdown button as suggested in #708.

So here's how notices could look:

![notices](https://cloud.githubusercontent.com/assets/1204802/26671288/292f7da4-46b5-11e7-8c5d-353437eb54dd.png)

**Comment by afercia** [comment]
Worth noting that all these notices would need an equivalent audible feedback for screen reader users. As a general rule: any "visual" notice needs to be communicated also non-visually.

**Comment by melchoyce** [comment]
FWIW, we have a separate set of notice styles in core for notices on a white background:

![image](https://cloud.githubusercontent.com/assets/2846578/26700910/21e44164-46ed-11e7-90aa-27d93d41abc6.png)
![image](https://cloud.githubusercontent.com/assets/2846578/26700943/3db0627e-46ed-11e7-9104-a31ae32c4e87.png)
![image](https://cloud.githubusercontent.com/assets/2846578/26700950/44abde46-46ed-11e7-9159-b949a40e0905.png)

If the inspector/sidebar is open, would the notice overlap or appear to the left of it?

**Comment by jasmussen** [comment]
Let's totally use the colored versions as they are.

**Comment by melchoyce** [comment]
Cool — they all use the `.notice-alt` class: 

```
<div class="update-message notice inline notice-warning notice-alt">
    <p>Notice Message.</p>
</div>
```
